src/calculate-fps.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+const {max} = Math
+
+type Frame = number
+type VisibilityEvent = [boolean, number]
+
+export default (frames: Array<Frame>, visibilityEvents: Array<VisibilityEvent>) => {
+  const totalTime = frames[frames.length - 1] - frames[0]
+  const visibleTime = totalTime - getTotalHiddenTime(visibilityEvents)
+  const amountOfFrames = max(frames.length - 1, 1)
+  const averageFrameDuration = visibleTime / amountOfFrames
+  const framesPerSecond = 1000 / averageFrameDuration
+
+  return framesPerSecond
+}
+
+export const getTotalHiddenTime = (visibilityEvents: Array<VisibilityEvent>) =>
+  visibilityEvents
+    .reduce((ranges, [visible, timestamp]) => {
+      // We only care about periods when the window was hidden, so
+      // if the first event is making the window visible, we ignore it
+      if (visible && ranges.length === 0) {
+        return ranges
+      }
+
+      const last = ranges[ranges.length - 1]
+      const remaining = ranges.slice(0, -1)
+
+      return !visible
```

Comment:
![deal-with-it](https://cloud.githubusercontent.com/assets/6001/21985101/7b10e314-dbf9-11e6-964d-cd9ae8e480de.jpg)